src/fparser/two/tests/fortran2003/test_specific_binding_r451.py

```diff
@@ -0,0 +1,180 @@
+
+    obj = testcls('procedure, pass :: length => point_length')
+    assert isinstance(obj, testcls), repr(obj)
+    assert str(obj) == "PROCEDURE, PASS :: length => point_length"
+
+    obj = testcls(lines[11].strip())
+    assert isinstance(obj, testcls), repr(obj)
+    assert str(obj) == "PROCEDURE :: internal_method_1"
+
+    obj = testcls(lines[12].strip())
+    assert isinstance(obj, testcls), repr(obj)
+    assert str(obj) == "PROCEDURE internal_method_2"
+
+    obj = testcls(lines[13].strip())
+    assert isinstance(obj, testcls), repr(obj)
+    assert str(obj) == "PROCEDURE, PUBLIC :: submethod_1"
+
+
+def test_binding_pass_arg_name(f2003_create):
+    ''' Test that Binding_PASS_Arg_Name is parsed correctly. '''
+
+    testcls = Binding_PASS_Arg_Name
+    parsecls = Specific_Binding
+
+    aobj = testcls("pass(name)")
+    assert isinstance(aobj, testcls), repr(aobj)
+    assert repr(aobj) == ("Binding_PASS_Arg_Name('PASS', Name('name'))")
+
+    line = ("procedure(analytic_function), deferred, pass(self) :: "
+            "function_name")
+    aobj = parsecls(line).items[1].items[1]
+    assert isinstance(aobj, testcls), repr(aobj)
+    assert repr(aobj) == "Binding_PASS_Arg_Name('PASS', Name('self'))"
+
+    line = SOURCE.split('\n')[14].strip()
+    obj = parsecls(line)
+    assert isinstance(obj, parsecls), repr(obj)
+    assert (str(obj) == "PROCEDURE, PUBLIC, PASS(self) :: "
+            "abstract_method => implementation_method")
+    assert repr(obj) == (
+        "Specific_Binding(None, "
+        "Binding_Attr_List(',', (Access_Spec('PUBLIC'), "
+        "Binding_PASS_Arg_Name('PASS', Name('self')))), "
+        "'::', Name('abstract_method'), Name('implementation_method'))")
+
+
+def test_specific_binding_error(f2003_create):
+    ''' Test that parsing invalid Fortran syntax for
+    Specific_Binding statements raises an appropriate error. '''
+    testcls = Specific_Binding
+
+    with pytest.raises(NoMatchError) as excinfo:
+        _ = testcls('procedure populate_entity => populate_cube')
+    assert ("Specific_Binding: 'procedure populate_entity => "
+            "populate_cube'" in str(excinfo))
+
+    with pytest.raises(NoMatchError) as excinfo:
+        _ = testcls('procedure :: populate_entity =>')
+    assert ("Specific_Binding: 'procedure :: populate_entity =>"
+            in str(excinfo))
+
+    with pytest.raises(NoMatchError) as excinfo:
+        _ = testcls('procedure, :: sub_gen => sub_spec')
+    assert ("Specific_Binding: 'procedure, :: sub_gen => sub_spec'"
+            in str(excinfo))
+
+    line = SOURCE.split('\n')[13].strip()
+    line = line.replace(" :: ", " ")
+    with pytest.raises(NoMatchError) as excinfo:
+        _ = testcls(line)
+    assert ("Specific_Binding: 'procedure, public submethod_1'"
+            in str(excinfo))
+
+
+def test_specific_binding_fserror(f2003_create):
+    ''' Test that an error in Specific_Binding within a program
```

Comment:
I don't think we need to test for a parent class raising a syntax error. I think we can stick within this class here.

Reply:
Please add explicit tests for c456 and c457. Please also specify that c458 can't be tested with this parser - or perhaps add an xfail test for future reference